addons/web/controllers/home.py

```diff
@@ -38,7 +38,7 @@ def _web_client_readonly(self, registry, request):
         return False
 
     # ideally, this route should be `auth="user"` but that don't work in non-monodb mode.
-    @http.route(['/web', '/odoo', '/odoo/<path:subpath>'], type='http', auth="none", readonly=_web_client_readonly)
+    @http.route(['/web', '/odoo', '/odoo/<path:subpath>', '/scoped_app/<path:subpath>'], type='http', auth="none", readonly=_web_client_readonly)
```

Comment:
This is the primary entrypoint to the web client. It also is non-trivial due to the shenanigans regarding `auth='none'` that actually should behave like a `auth='public'`. I have a strong feeling that we shouldn't offer new routes to this specific controller. Would you like to explain me why this specific controller need a new route? Also that it is identical to `/odoo`, why not use that one? Or why not define a dedicated controller?

Reply:
Since there is already a PWA covering the scope `/odoo`, it's basically impossible to handle PWA with scopes such as `/odoo/barcode ` `/odoo/field-service` without interfering with the main app. The navigators don't properly support scopes using subpaths. They might not open the right app, or prompt the installation of an app if there is already one being installed covering a much larger scope.
-> example: I actually tested it, you install Odoo app then it is not possible to install Barcode with a scope /odoo/barcode

Also, we look for the current URL to know if the app is being accessed from the navigator, the main PWA or the specific PWA. This allows the UX to be slightly adapted (no back to home  menu button, or button to install the app)

Reply:
Thanks for the explanation :)